### ISSUE TITLE
Trait with two responses causes exception during code generation.

### ISSUE THREAD

**Issue by milan-janecek** [mulesoft-labs/raml-dotnet-tools#37]
The RAML file in the picture causes exception during code generation. Seems like key is not correctly set in ParseTraitsResponses method of GeneratorServiceBase class.

Please check this, RAML spec allows having trait with 2 responses.
![trait-with-2-responses](https://cloud.githubusercontent.com/assets/13526850/10497932/082749d2-72c9-11e5-946e-749073b749cc.PNG)

**Comment by woodp** [comment]
the RAML is missing a colon after the "is" (in the last line)
if I add that line the RAML parses fine

**Comment by milan-janecek** [comment]
yes, there was a mistake, but even if I correct it, visual studio still shows error during code generation.

Other raml file and exception from ActivityLog is attached, please, reconsider.

![trait-with-2-responses-v2](https://cloud.githubusercontent.com/assets/13526850/10517409/ea682b20-735c-11e5-986e-f69b7ca6554e.PNG)
![activity-log-exception](https://cloud.githubusercontent.com/assets/13526850/10517414/ee730f46-735c-11e5-9800-b42ca5376bb1.PNG)

**Comment by woodp** [comment]
You are using the same id for both schemas, please change the id on one of them and try again.

**Comment by milan-janecek** [comment]
I added "id": "myId1" to first, "id": "myId2" to second. Same problem. I really believe it is a bug, RAML designer has no problems with that API definition, according to spec, it is a valid definition. I experimented a little and if I have 2 response codes directly under resource it works, if I have 2 response codes under resourceType it also works. It does not work under traits.

Is there still something I miss? I'd like this to be resolved as I want to declare traits for common response codes like 400 and 404.
